pkg/client/tx/client.go

```diff
@@ -240,8 +250,24 @@ func (txnClient *txClient) SignAndBroadcast(
 	timeoutHeight := txnClient.blockClient.LastBlock(ctx).
 		Height() + txnClient.commitTimeoutHeightOffset
 
-	// TODO_TECHDEBT: this should be configurable
-	txBuilder.SetGasLimit(690000042)
+	// Coin multiplication prevents doing it using a zero value.
+	if gasLimit > 0 {
+		txBuilder.SetGasLimit(gasLimit)
+
+		gasLimitDec := math.LegacyNewDec(int64(gasLimit))
```

Comment:
Why aren't we using `math.NewDec` here?

Reply:
There's no `NewDec`, even `DecCoins` use `LegacyDec` :face_with_diagonal_mouth: